client/components/suggestions/index.jsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { findIndex, isEqual, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Item from './item';
+
+class Suggestions extends Component {
+
+	static propTypes = {
+		query: PropTypes.string,
+		suggestions: PropTypes.arrayOf( PropTypes.shape( {
+			label: PropTypes.string,
+		} ) ).isRequired,
+		suggest: PropTypes.func.isRequired,
+	}
+
+	static defaultProps = {
+		query: '',
+	}
+
+	state = {
+		suggestionPosition: 0,
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( isEqual( nextProps.suggestions, this.props.suggestions ) ) {
+			return;
+		}
+
+		this.setState( { suggestionPosition: 0 } );
+	}
+
+	suggestionsCount = () => this.props.suggestions.length
+
+	getPositionForSuggestion = label => findIndex( this.props.suggestions, { label } )
+
+	suggest = position => this.props.suggest( this.props.suggestions[ position ] );
+
+	increasePosition = () => this.setState( {
+		suggestionPosition: ( this.state.suggestionPosition + 1 ) % this.suggestionsCount(),
+	} )
+
+	decreasePosition = () => this.setState( {
+		suggestionPosition: ( this.suggestionsCount() + this.state.suggestionPosition - 1 ) % this.suggestionsCount(),
+	} )
+
+	handleKeyEvent = ( event ) => {
+		if ( this.suggestionsCount() === 0 ) {
+			return;
+		}
+
+		switch ( event.key ) {
+			case 'ArrowDown':
+				this.increasePosition();
+				event.preventDefault();
+				break;
+
+			case 'ArrowUp':
+				this.decreasePosition();
+				event.preventDefault();
+				break;
+
+			case 'Enter':
+				this.state.suggestionPosition >= 0 && this.props.suggest( this.state.suggestionPosition );
+				break;
+		}
+	}
+
+	handleMouseDown = label => this.props.suggest( this.getPositionForSuggestion( label ) )
+
+	handleMouseOver = label => this.setState( {
+		suggestionPosition: this.getPositionForSuggestion( label ),
+	} )
+
+	render() {
+		const { query } = this.props;
+		const showSuggestions = this.suggestionsCount() > 0;
+
+		return (
+			<div>
```

Comment:
If there are no suggestions, this would render an empty `<div>`. Ideally, the check for whether or not `this.props.suggestions` is empty would happen in its parent, and the `Suggestions` component itself wouldn't be rendered at all in that case.

Reply:
The idea here was, that since the parent has to use `ref` on `Suggestions` to pass `onKeyDown` events, it might be better for it to just stay there ?

Reply:
It still feels a bit off to render a component that we know in the end will end up rendering nothing. We could do something like add a check that `this.refs.suggestions` is defined before trying to access it.

Or would the `SearchCard` component not be rendered either if there are no suggestions? In which case we wouldn't really need that check because the function wouldn't be hooked up to anything in the first place.